### DISCUSSION TITLE
can not build under ubuntu 20.04

### DISCUSSION THREAD

**Discussion by skydig** [rustdesk/rustdesk#4244] [General]
Package ffigen is currently active at version 5.0.1.
The package ffigen is already activated at newest available version.
To recompile executables, first run `dart pub global deactivate ffigen`.
Activated ffigen 5.0.1.
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
Resolving dependencies... (16.3s)
Because every version of flutter_localizations from sdk depends on intl 0.18.1 and every version of dash_chat_2 depends on intl ^0.17.0, flutter_localizations from sdk is incompatible with dash_chat_2.
So, because flutter_hbb depends on both flutter_localizations from sdk and dash_chat_2 ^0.0.14, version solving failed.
    Blocking waiting for file lock on build directory

   Compiling rustdesk v1.2.0 (/home/abc/rustdesk)
error: failed to run custom build command for `rustdesk v1.2.0 (/home/abc/rustdesk)`

Caused by:
  process didn't exit successfully: `/home/abc/rustdesk/target/debug/build/rustdesk-6ba66e2e394a434a/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=Cargo.toml
  cargo:rerun-if-changed=src/flutter_ffi.rs

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ffigen failed:
  stderr: Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
  Cannot open file, path = '.dart_tool/package_config.json' (OS Error: No such file or directory, errno = 2)

**is  build under ubuntu 20.04  supported?**

**Comment by SergeyMy** [comment]
for intl^0.17.0,
flutter must be version 3.8.0-6.0.pre or lower
//https://github.com/flutter/flutter/blob/3.8.0-6.0.pre/dev/tools/pubspec.yaml

**Comment by SergeyMy** [comment]
checked! builds from flutter-3.7.0 to flutter-3.7.12
